wikimedia/wikipedia-ios
# Watchlist API calls

**Phabricator:** 
https://phabricator.wikimedia.org/T334975

### Notes
This PR adds the necessary logic needed for the Watchlist API calls in the `WKData` repo. It implements the `MediaWikiNetworkService`, as well as coordinates app languages to the `WKDataEnvironment`.

### Test Steps

To test live watchlist calls, add `import Components` to the top of `ExploreViewController`, then in `viewDidAppear` add:

```
let viewModel = WKWatchlistViewModel()
let watchlist = WKWatchlistViewController(viewModel: viewModel)
present(watchlist, animated: true)
```

1. If you launch while you are logged in, upon Explore appearance you should see WKWatchlistViewController presented with data from your watchlist.
2. Explore the different action options while proxying, and confirm the calls look good.
3. Edit app languages, return back to the watchlist presentation and confirm the watchlist item fetch calls look accurate in Proxyman. We should always fetch watchlist items from app languages + commons + wikidata domains. There is no integrated API call so they need to be made separately.

Note: We should first merge https://github.com/wikimedia/wikipedia-ios-components/pull/11, then point the Components package back to the `main` branch, before merging this one.
